cmd/metal-api/internal/service/v1/s3.go

```diff
@@ -0,0 +1,53 @@
+package v1
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type S3Client struct {
+	*s3.Client
+	Region string
+	Url    string
+	Key    string
+	Secret string
+}
+
+func NewS3Client(region, url, key, secret string) *S3Client {
+	return &S3Client{
+		Region: region,
+		Url:    url,
+		Key:    key,
+		Secret: secret,
+	}
+}
+
+func (c *S3Client) Connect() error {
+	if c.Client != nil {
+		return nil
+	}
+
+	customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+		if c.Region == region {
+			return aws.Endpoint{
+				PartitionID:       "aws",
+				URL:               c.Url,
+				SigningRegion:     region,
+				HostnameImmutable: true,
+			}, nil
+		}
+		// returning EndpointNotFoundError will allow the service to fallback to it's default resolution
+		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
+	})
```

Comment:
I guess you tried this against a GCP Bucket? For this, I think we have simpler connection params that we use for the image-cache-sync: https://github.com/metal-stack/metal-image-cache-sync/blob/master/cmd/main.go#L175-L191